src/Compilers/Core/Portable/Diagnostic/WellKnownDiagnosticTags.cs

```diff
@@ -38,6 +38,18 @@ public static class WellKnownDiagnosticTags
         /// </summary>
         public const string NotConfigurable = nameof(NotConfigurable);
 
+        /// <summary>
+        /// Indicates that the analyzer reporting the diagnostic supports custom severity configuration mechanism(s)
+        /// to allow end users to configure effective severity of the diagnostic.
+        /// Such analyzers are always considered to be enabled by the compiler and always receive analyzer callbacks.
+        /// Additionally, severity of the diagnostics reported with this custom tag is not altered by analyzer config options
+        /// to configure severity, i.e. 'dotnet_diagnostic' and 'dotnet_analyzer_diagnostic' entries. 
+        /// </summary>
+        /// <remarks>
+        /// See https://github.com/dotnet/roslyn/issues/52991 for further details.
+        /// </remarks>
+        public const string CustomConfigurable = nameof(CustomConfigurable);
```

Comment:
@sharwell This is the crux of the design that will ensure callbacks are always made into such analyzers, regardless of their default severity or any compiler-known severity configuration entries. We don't need the user project to have any dummy, blank source file with severities bumped to warning/error to ensure callbacks are always executed. The same tag is also used to ensure that `dotnet_diagnostic` and `dotnet_analyzer_diagnostic` entries are not applied to these diagnostics.